src/semidiscretization/semidiscretization_hyperbolic.jl

```diff
@@ -44,16 +44,30 @@ Construct a semidiscretization of a hyperbolic PDE.
 """
 function SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver;
                                       source_terms=nothing,
-                                      boundary_conditions=boundary_condition_periodic, RealT=real(solver))
+                                      boundary_conditions=boundary_condition_periodic,
+                                      RealT=real(solver), uEltype=RealT)
 
-  cache = create_cache(mesh, equations, solver, RealT)
+  cache = create_cache(mesh, equations, solver, RealT, uEltype)
   _boundary_conditions = digest_boundary_conditions(boundary_conditions)
 
   SemidiscretizationHyperbolic{typeof(mesh), typeof(equations), typeof(initial_condition), typeof(_boundary_conditions), typeof(source_terms), typeof(solver), typeof(cache)}(
     mesh, equations, initial_condition, _boundary_conditions, source_terms, solver, cache)
 end
 
 
+# Create a new semidiscretization but change some parameters compared to the input.
+# `Base.similar` follows a related concept but would require us to `copy` the `mesh`,
+# which would impact the performance. Instead, `SciMLBase.remake` has exactly the
+# semantics we want to use here. In particular, it allows us to re-use mutable parts,
+# e.g. `remake(semi).mesh === semi.mesh`.
+function remake(semi::SemidiscretizationHyperbolic; RealT=real(semi.solver), uEltype=RealT)
```

Comment:
This only allows to change `RealT` and `uEltype`. Is this intended? Otherwise, couldn't we use something like the following:
```suggestion
function remake(semi::SemidiscretizationHyperbolic; mesh=semi.mesh, equations=semi.equations, ..., RealT=real(semi.solver), uEltype=RealT)
```

Reply:
I removed the `RealT` here, since we don't use it right now and we should think a bit more about consistency when using it. For example, shall we also update the `RealT` of the `equations` when calling this method of `remake` here?

Reply:
No, since we are calling `remake` on a `semi`, not on the `equatiosn`. We could also add a `remake` method for the `equations`, but that would be a separate call (I think).

Reply:
But then the default should be `real(semi)`, which falls back to `real(semi.solver)`, which is not necessarily consistent with `RealT` used in `semi.cache`. How shall we handle that?

Reply:
Ah, right. In that case, go head and just allow to change `uEltype` :+1: